BlockchainSdk/WalletManagers/Chia/Common/ClvmProgram.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  ClvmProgram.swift
+//  BlockchainSdk
+//
+//  Created by skibinalexander on 11.07.2023.
+//  Copyright © 2023 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import TangemSdk
+import CryptoSwift
+
+class ClvmNode {
+    private let atom: Array<Byte>?
+    private let left: ClvmNode?
+    private let right: ClvmNode?
+
+    init(atom: Array<Byte>? = nil, left: ClvmNode? = nil, right: ClvmNode? = nil) {
+        self.atom = atom
+        self.left = left
+        self.right = right
+    }
+    
+    // MARK: - Hashable
+    
+    func hash() throws -> Data {
+        let hash: Data
+        
+        if let value = atom {
+            hash = Data([1] + value).sha256()
+        } else {
+            if let left = try left?.hash(), let right = try right?.hash() {
+                hash = Data([2] + left + right).sha256()
+            } else {
+                throw NSError()
+            }
+        }
+        
+        return hash
+    }
```

Comment:
```suggestion
    func hash() throws -> Data {
        if let value = atom {
            return Data([1] + value).sha256()
        } else if let left = try left?.hash(), let right = try right?.hash() {
            return Data([2] + left + right).sha256()
        }
        
        throw NSError()
    }
```

Так попроще как будто)

Reply:
> Так попроще как будто)

Поправил, дебажил, поэтому пришлось на условия разбить